Normalize image color spaces before comparison

ejensen opened the pull request:
## Overview
Use the sRGB converted snapshot when doing the perceptual comparison. This reduces the chances of failures when comparing snapshots using different color spaces.

This colorspace normalization technique was originally introduced in https://github.com/pointfreeco/swift-snapshot-testing/pull/446 and this PR extends it to both the reference and new images when performing perceptual image comparison.

Unit tests were added to verify that images in the P3 and sRGB colors spaces match after colorspace normalization.

## Related Issues
* https://github.com/pointfreeco/swift-snapshot-testing/issues/313#issuecomment-1258535742

pilot34 commented:
Hey. Should we merge this PR after the rebase? Looks like it was approved, but never merged

Kaspik commented:
@stephencelis Please, is there a plan to merge this?

ogiba commented:
@stephencelis I will bump this topic cause there was no answer from few months

Could you provide information what is blocking you from merging it into the main branch?

It will be nice feature/fix that could reduce lots of issues.

Thanks!

brucerune commented:
+1 please merge :)

ejensen commented:
Hi @mbrandonw or @stephencelis, would one of you be able to look at this PR? It has been waiting for a while and addresses issues when collaborating with people using various hardware to run snapshot tests. At least six other people (unaffiliated with me) stated they are experiencing these issues.

AndreasInk commented:
I'm also experiencing this issue, would love to see this merged @stephencelis @mbrandonw

jiayangmiao commented:
Any chance to check on conflicts and merge these? I am experiencing these issues too.
